agimus_controller_ros/hpp_agimus_controller.py

```diff
@@ -1,121 +1,21 @@
-            get_last_joint(self.rmodel)
-        )
-        self.nq = self.rmodel.nq
-        self.nv = self.rmodel.nv
-        self.nx = self.nq + self.nv
         self.hpp_interface = HppInterface()
-        self.armature = np.array([0.05] * self.nq)
-
-        self.ocp = OCPCrocoHPP(
-            self.rmodel, self.cmodel, use_constraints=False, armature=self.armature
-        )
-        self.ocp.set_weights(10**4, 10, 10**-3, 0)
-        self.mpc_iter = 0
-        self.save_predictions_and_refs = False
-        self.nb_mpc_iter_to_save = None
-        self.mpc_data = {}
-
-        self.rate = rospy.Rate(self.params.rate, reset=True)
-        self.mutex = Lock()
-        self.sensor_msg = Sensor()
-        self.control_msg = Control()
-        self.ocp_solve_time = Duration()
-        self.x0 = np.zeros(self.nq + self.nv)
-        self.x_guess = np.zeros(self.nq + self.nv)
-        self.u_guess = np.zeros(self.nv)
-        self.state_subscriber = rospy.Subscriber(
-            "robot_sensors",
-            Sensor,
-            self.sensor_callback,
-        )
-        self.control_publisher = rospy.Publisher(
-            "motion_server_control", Control, queue_size=1, tcp_nodelay=True
-        )
-        self.ocp_solve_time_pub = rospy.Publisher(
-            "ocp_solve_time", Duration, queue_size=1, tcp_nodelay=True
-        )
-        self.start_time = 0.0
-        self.first_robot_sensor_msg_received = False
-        self.first_pose_ref_msg_received = True
-
-    def sensor_callback(self, sensor_msg):
-        with self.mutex:
-            self.sensor_msg = deepcopy(sensor_msg)
-            if not self.first_robot_sensor_msg_received:
-                self.first_robot_sensor_msg_received = True
-
-    def wait_first_sensor_msg(self):
-        wait_for_input = True
-        while not rospy.is_shutdown() and wait_for_input:
-            wait_for_input = (
-                not self.first_robot_sensor_msg_received
-                or not self.first_pose_ref_msg_received
-            )
-            if wait_for_input:
-                rospy.loginfo_throttle(3, "Waiting until we receive a sensor message.")
-                with self.mutex:
-                    sensor_msg = deepcopy(self.sensor_msg)
-                    self.start_time = sensor_msg.header.stamp.to_sec()
-            rospy.loginfo_once("Start controller")
-            self.rate.sleep()
-        return wait_for_input
-
-    def wait_buffer_has_twice_horizon_points(self):
-        while (
-            self.traj_buffer.get_size(self.point_attributes)
-            < 2 * self.params.horizon_size
-        ):
-            self.fill_buffer()
-
-    def fill_buffer(self):
-        point = self.get_next_trajectory_point()
-        if point is not None:
-            self.traj_buffer.add_trajectory_point(point)
+        self.plan_is_set = False
 
     def get_next_trajectory_point(self):
+        if not self.plan_is_set:
```

Comment:
Why not set plan in the constructor?
Is it because you need the current robot state?

Reply:
because we need the robot state yes